File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/enums/RegisterCenterType.java
package org.javamaster.invocationlab.admin.enums;

/**
 * @author yudong
 * @date 2022/11/8
 */
public enum RegisterCenterType {
    ZK(1),
    EUREKA(2),
    ;

    public final int type;

    RegisterCenterType(int type) {
        this.type = type;
    }

    public static RegisterCenterType getByType(Integer type) {
        if (type == null) {
            return ZK;
        }
        for (RegisterCenterType value : RegisterCenterType.values()) {
            if (value.type == type) {
                return value;
            }
        }
        return null;
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/enums/RegisterCenterType.kt
package org.javamaster.invocationlab.admin.enums

/**
 * @author yudong
 * @date 2022/11/8
 */
enum class RegisterCenterType(@JvmField val type: Int) {
    ZK(1),
    EUREKA(2),
    ;

    companion object {
        @JvmStatic
        fun getByType(type: Int?): RegisterCenterType? {
            if (type == null) {
                return ZK
            }
            for (value in entries) {
                if (value.type == type) {
                    return value
                }
            }
            return null
        }
    }
}
